src/Avalonia.Styling/ClassBindingManager.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Collections.Generic;
+using Avalonia.Data;
+
+namespace Avalonia
+{
+    internal static class ClassBindingManager
+    {
+        private static readonly Dictionary<string, AvaloniaProperty> s_RegisteredProperties =
+            new Dictionary<string, AvaloniaProperty>();
+        
+        public static IDisposable Bind(IStyledElement target, string className, IBinding source, object anchor)
+        {
+            if (!s_RegisteredProperties.TryGetValue(className, out var prop))
```

Comment:
Should we care about clearing this dictionary on binding disposed? To not keep unused class-properties in the memory?
I doubt if it will be any problem though.

Reply:
I don't think it's possible to unregister AvaloniaProperty